legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/auth/GitLabTokenManager.java

```diff
@@ -90,12 +94,45 @@ void setGitLabToken(GitLabToken token)
         this.token = token;
     }
 
+    public void setRefreshToken(GitLabToken refreshToken)
+    {
+        if (refreshToken == null)
+        {
+            throw new IllegalArgumentException("token may not be null");
+        }
+        if (refreshToken.getTokenType() == null)
+        {
+            throw new IllegalArgumentException("token type may not be null");
+        }
+        this.refreshToken = refreshToken;
+    }
+
+    public void setTokenExpiry(long expiresIn)
+    {
+        if (expiresIn == 0)
+        {
+            expiresIn = DEFAULT_EXPIRY;
+        }
+        this.tokenExpiry = LocalDateTime.now().plusSeconds(expiresIn -  1800);
+    }
```

Comment:
Let's make clear we're talking about seconds. Also, what if the expiry is close to or less than our default threshold? We might want to use a lower threshold in that case. (Maybe just divide the expiry in half or something?)
```suggestion
    public void setTokenExpiry(long expiresInSecs)
    {
        if (expiresIn <= 0)
        {
            expiresIn = DEFAULT_EXPIRY_SECS;
        }
        this.tokenExpiry = LocalDateTime.now().plusSeconds(expiresIn - DEFAULT_REFRESH_THRESHOLD_SECS);
    }
```

Reply:
maybe we can do a three quarters than half, so we don't hit the gitlab server to get more tokens

Reply:
That seems reasonable.